src/content/docs/en/basics/astro-components.mdx

```diff
@@ -318,6 +318,40 @@ const { title } = Astro.props;
 </div>
 ```
 
+#### Fallback Content for Empty Slots
+
+:::note
+Fallback content will not show if an empty slot is passed into a component.
+:::
+
+```astro title="ComponentA.astro"
+<slot name="foo">
+  <p>Default Content</p>
+</slot>
+```
+
+```astro title="ComponentB.astro"
+---
+import ComponentA from ""./ComponentA.astro";
+---
+
+<ComponentA>
+  <slot name="bar" slot="foo" />
+</ComponentA>
+```
+
+```astro title="index.astro"
+---
+import ComponentB from ""./ComponentB.astro";
+---
+
+<ComponentB />
+```
+
+Here we are not passing a slot into `ComponentB` from `index.astro`. But `ComponentB` is still passing an empty slot into `ComponentA`. Therefore `ComponentA` will not render it's default content.
+
+This is because a component checks that a slot exists - not if it's empty.
```

Comment:
```suggestion
Fallback content will only be displayed when no matching elements with the `slot="name"` attribute are found to pass into a named slot. 

If an empty slot (a slot that exists but contains no content) is passed into a component, then the fallback content will not be displayed.
```
I'm wondering whether we can get away with this, and not showing an example here. Would you say this explanation is correct? And if so, do you think it's *sufficient* without showing three different code blocks?

I would argue that if it's correct, it's NWTWWHB! And, if we get questions/complaints, we can always add code snippets in later. What do you think?

Reply:
@sarah11918 Thank you! 🙌 

I think that explanation is pretty perfect and concise! The only thing I'm thinking is: it's a fairly uncommon situation to be in. I think the code snippet sort of highlights that really you should never really have this scenario in your code - but ofc it _will_ happen to someone eventually.

Also I feel people, myself included, would assume that an empty slot _wouldn't_ pass an empty slot. So I feel like this section is less about telling people that fallback content will only displayed when blah blah blah. It's more about informing people that it _is_ in fact possible to pass an empty slot to a child. I would've assumed the slot would _only_ get passed if it contains content, which clearly is not the case.

I'm more than happy to have just this explanation for now and get feedback before adding more to it. I feel it's a very uncommon use case, and at least those unlucky enough to run into this can see this bit here immediately with the rest of the slots documentation.

Reply:
OK, if it's less about the fallback then we could shift the second sentence to something like...??

> Astro will pass an **empty slot** when a slot element exists but has no content to pass, and fallback content will not be displayed. Therefore, fallback content cannot be used as a default or check for a case where no content is passed. Fallback content is only displayed when no slot element can be found.

Reply:
Yeah I think that makes more sense tbh from that angle! But it is still quite complex to explain! What about:

> Astro will pass an **empty** slot when a slot element exists but has no content to pass. Fallback content cannot be used as a default when an empty slot is passed. Fallback content is _only_ displayed when no slot element can be found.

Reply:
Ooh, even simpler! I like it! Ship it!